src/main/java/appeng/core/RecipeLoader.java

```diff
@@ -44,56 +44,68 @@
 	{
-		Preconditions.checkNotNull( handler );
-
-		this.handler = handler;
+		this.recipeDirectory = Preconditions.checkNotNull( recipeDirectory );
+		Preconditions.checkArgument( !recipeDirectory.isFile() );
+		this.config = Preconditions.checkNotNull( config );
+		this.handler = Preconditions.checkNotNull( handler );
 	}
 
 	@Override
-	public void run()
+	public final void run()
 	{
-		// setup copying
-		final RecipeResourceCopier copier = new RecipeResourceCopier( "assets/appliedenergistics2/recipes/" );
-		final File configDirectory = AppEng.instance().getConfigDirectory();
-		final File generatedRecipesDir = new File( configDirectory, "generated-recipes" );
-		final File userRecipesDir = new File( configDirectory, "user-recipes" );
-		final File readmeGenDest = new File( generatedRecipesDir, "README.html" );
-		final File readmeUserDest = new File( userRecipesDir, "README.html" );
-
-        // generates generated and user recipes dir
-		// will clean the generated every time to keep it up to date
-		// copies over the recipes in the jar over to the generated folder
-		// copies over the readmes
-		try
+		if( this.config.isEnabled() )
 		{
-			FileUtils.forceMkdir( generatedRecipesDir );
-			FileUtils.forceMkdir( userRecipesDir );
-			FileUtils.cleanDirectory( generatedRecipesDir );
+			// setup copying
+			final RecipeResourceCopier copier = new RecipeResourceCopier( "assets/appliedenergistics2/recipes/" );
 
-			copier.copyTo( generatedRecipesDir );
-			FileUtils.copyFile( readmeGenDest, readmeUserDest );
+			final File generatedRecipesDir = new File( this.recipeDirectory, "generated" );
+			final File userRecipesDir = new File( this.recipeDirectory, "user" );
 
-			// parse recipes prioritising the user scripts by using the generated as template
-			this.handler.parseRecipes( new ConfigLoader( generatedRecipesDir, userRecipesDir ), "index.recipe" );
-		}
-		// on failure use jar parsing
-		catch( IOException e )
-		{
-			AELog.error( e );
-			this.handler.parseRecipes( new JarLoader( "/assets/appliedenergistics2/recipes/" ), "index.recipe" );
+			// generates generated and user recipes dir
+			// will clean the generated every time to keep it up to date
+			// copies over the recipes in the jar over to the generated folder
+			// copies over the readmes
+			try
+			{
+				FileUtils.forceMkdir( generatedRecipesDir );
+				FileUtils.forceMkdir( userRecipesDir );
+				FileUtils.cleanDirectory( generatedRecipesDir );
+
+				copier.copyTo( ".recipe", generatedRecipesDir );
+				copier.copyTo( ".html", recipeDirectory );
+
+				// parse recipes prioritising the user scripts by using the generated as template
+				this.handler.parseRecipes( new ConfigLoader( generatedRecipesDir, userRecipesDir ), "index.recipe" );
+			}
+			// on failure use jar parsing
+			catch( IOException e )
+			{
+				AELog.error( e );
+				this.handler.parseRecipes( new JarLoader( "/assets/appliedenergistics2/recipes/" ), "index.recipe" );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/assets/appliedenergistics2/recipes/" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)